gpt_bpe.go

```diff
@@ -366,6 +366,19 @@ func NewEncoder(vocabId string) (*GPTEncoder, error) {
 		tokenizerSpecialConfig.AddBosToken = true
 		tokenizerSpecialConfig.AddEosToken = true
 	}
+
+	// Add in default pad token if not already set
+	if tokenizerSpecialConfig.PadToken == "" && hfConfig.PadTokenStr == nil {
+		// Inject the pad token into the encoder to uintmax16,
+		// throw an error if vocab is larger than uintmax16
+		if len(encoderTokens) >= math.MaxInt16 {
+			log.Fatalf("Vocab size is larger than uint16 max, default pad token cannot be added." +
+				"Please specify a pad token in the vocab file.")
+		}
+		encoderTokens["[PAD]"] = math.MaxUint16
+		tokenizerSpecialConfig.PadToken = "[PAD]"
```

Comment:
Is there a way to configure what the default padding string would be if there is none?

Reply:
currently no, hardcoded. Every encoder object is defined with just the vocabID. I could technically add it as an arg here but I think it would cause some issues.
```func NewGPT2Encoder() GPTEncoder {
	encoder, _ := NewEncoder(VOCAB_ID_GPT2)
	return *encoder
}```